src/main/java/org/mqttbee/api/mqtt/MqttClientBuilder.java

```diff
@@ -53,20 +57,36 @@ public MqttClientBuilder withIdentifier(@NotNull final MqttClientIdentifier iden
     }
 
     @NotNull
-    public MqttClientBuilder forServerHost(@NotNull final String host) {
+    public MqttClientBuilder withServerHost(@NotNull final String host) {
         this.serverHost = host;
         return this;
     }
 
     @NotNull
-    public MqttClientBuilder forServerPort(final int port) {
+    public MqttClientBuilder withServerPort(final int port) {
         this.serverPort = port;
         return this;
     }
 
     @NotNull
-    public MqttClientBuilder usingSSL(final boolean usesSSl) {
-        this.usesSSL = usesSSl;
+    public MqttClientBuilder usingSsl() {
+        return usingSsl(MqttClientSslConfigImpl.DEFAULT);
+    }
+    
+    @NotNull
+    public MqttClientBuilder usingSsl(@NotNull final MqttClientSslConfig sslConfig) {
```

Comment:
Imho this should also be @Nullable as the WebsocketConfig (this removes the SslConfig)

Reply:
Hmm, should the client builder be reusable? So you can build multiple clients with the same builder?

If not, I would say NotNull is correct, because
* in case of not using SSL you would not call any of the usingSsl methods
* in case of using SSL with default config you would call usingSsl()
* in case of using SSL with custom config you would use usingSsl(customConfig)

If yes, then of course we need a way to reset the SSL config to null.

The same applies to the WebsocketConfig.

Reply:
I totally agree, then we should annotate both with @ NotNull

Reply:
DONE